nomenclature/code.py

```diff
@@ -61,11 +51,11 @@ def _replace_tags(code, tag, target_list):
     _code_list = []
 
     for target in target_list:
-        key = code.name.replace(tag, target.name)
+        key = code.name.replace("{" + tag + "}", target.name)
```

Comment:
Could use an f-string for that ;) but maybe with escaping the curly brackets the string concatenation is easier to read.
```suggestion
        key = code.name.replace(f"{{{tag}}}", target.name)
```